aten/src/TH/THTensor.cpp

```diff
@@ -160,5 +161,15 @@ void THTensor_stealAndSetStoragePtr(THTensor* tensor, THStorage* storage) {
   // Caffe2 might have tensors whose storages are null, but we
   // don't allow it in PyTorch.
   AT_ASSERT(storage);
+  // Caffe2 also has uninitialized dtype states, which we disallow here
```

Comment:
Caffe2 tensor won't make it so far because of constructor enforces. But it's a still valid check